README.md

```diff
@@ -39,11 +39,46 @@ d = debugger(argv=<"./test" | ["./test", ...]>,
     [auto_interrupt_on_command=<True | False>], #defaults to True
 )
 ```
-By setting `continue_to_binary_entrypoint` to False, the debugger will not automatically reach the entrypoint of the binary on `run()`, instead it will stop at the first instruction executed by the loader.
-By setting `auto_interrupt_on_command` to False, the debugger will wait for the debugged executable to script before issuing any command, while in the normal configuration it interrupts the executable in order to instantly issue commands, such as `breakpoint(...)` and any register access operation.
+By setting `continue_to_binary_entrypoint` to False, the `run()` command will stop at the first instruction executed by the loader instead of reaching the entrypoint of the binary.
+
+---
+
+The flag `auto_interrupt_on_command` fundamentally changes the way you use libdebug. By default it is set to True. In this setting, every debugging command transparenty stops the execution of the program to perform the requested action as soon as possible. This is an example extract of code in the default mode:
+
+```python
+d = debugger("./binary")
+
+bp = d.breakpoint("function")
+
+d.run()
+d.cont()
+
+# If you do not call d.wait() here, the register access will be performed
+# shortly after the process is allowed to continue
+d.wait()
+print(hex(d.rip))
+
+d.kill()
+```
+
+Instead, when set to False, issued commands will not be performed until a breakpoint is hit or any other tracing signal stops the process (e.g, SIGSEGV).
+
+```python
+d = debugger("./binary")
+
+bp = d.breakpoint("function")
+
+d.run()
+d.cont()
+
+# Here the register access is performed after the breakpoint is hit
+print(hex(d.rip))
+
+d.kill()
+```
 
 ## Register Access
-Registers are provided as properties of the class `Debugger`. You can read from and write to them when the process is interrupted.
+Registers are provided as properties of the debugger object. You can perform read and write operations on them, which by default are handled when the process is stopped by a breakpoint or another tracing signal.
```

Comment:
Well, registers are provided as properties of the ThreadContext. The Debugger object internally exposes as its own the properties of the first registered ThreadContext through a __getattr__ override. Should we say it?

Reply:
It is a README, not a full documentation. Let's keep it as simple as possible